WebApplication/Services/AdoptProjectService.cs

```diff
@@ -1,26 +1,54 @@
-﻿using Microsoft.Extensions.Logging;
+﻿using System.IO;
+using System.Net;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 using WebApplication.Definitions;
+using WebApplication.State;
 
 namespace WebApplication.Services
 {
     public class AdoptProjectService
     {
         private readonly ILogger<AdoptProjectService> _logger;
+        private readonly ProjectService _projectService;
 
-        public AdoptProjectService(ILogger<AdoptProjectService> logger)
+        public AdoptProjectService(ILogger<AdoptProjectService> logger, ProjectService projectService)
         {
-            this._logger = logger;
+            _logger = logger;
+            _projectService = projectService;
         }
 
         /// <summary>
         /// https://jira.autodesk.com/browse/INVGEN-45256
         /// </summary>
         /// <param name="payload">project configuration with parameters</param>
-        public void AdoptProjectWithParameters(AdoptProjectWithParametersPayload payload)
+        public string AdoptProjectWithParameters(AdoptProjectWithParametersPayload payload)
         {
             _logger.LogInformation($"adopting project {payload.Name}");
 
-            //TODO: add implementation
+            var localFileName = Path.GetTempFileName();
```

Comment:
You might want to use `TempFile`, which gets autodeleted on disposal:
https://github.com/Developer-Autodesk/forge-configurator-inventor/blob/master/WebApplication/Utilities/TempFile.cs